src/pal/tests/palsuite/file_io/GetFileSizeEx/test1/GetFileSizeEx.c

```diff
@@ -0,0 +1,214 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+/*=====================================================================
+**
+** Source:  GetFileSizeEx.c (test 1)
+**
+** Purpose: Tests the PAL implementation of the GetFileSizeEx function.
+**
+**
+**===================================================================*/
+
+#include <palsuite.h>
+
+const char* szTextFile = "text.txt";
+
+void CleanUp(HANDLE hFile)
+{
+    if (CloseHandle(hFile) != TRUE)
+    {
+        Fail("GetFileSizeEx: ERROR -> Unable to close file \"%s\".\n", 
+            szTextFile);
+    }
+    if (!DeleteFileA(szTextFile))
+    {
+        Fail("GetFileSizeEx: ERROR -> Unable to delete file \"%s\".\n", 
+            szTextFile);
+    }
+}
+
+
+int __cdecl main(int argc, char *argv[])
+{
+    HANDLE hFile = NULL;
+    BOOL bRc = FALSE;
+    DWORD dwRc = 0;
+    DWORD dwError = 0;
+    DWORD dwHighOrder = 0;
+    DWORD dwOffset = 0;
+    DWORD dwReturnedHighOrder = 0;
+    DWORD dwReturnedOffset = 0;
+    DWORD lpNumberOfBytesWritten;
+    LARGE_INTEGER qwFileSize;
+    LARGE_INTEGER qwFileSize2;
+    char * data = "1234567890";
+
+    qwFileSize.QuadPart = 0;
+    qwFileSize2.QuadPart = 0;
+
+    if (0 != PAL_Initialize(argc,argv))
+    {
+        return FAIL;
+    }
+
+
+    /* test on a null file handle */
+    bRc = GetFileSizeEx(hFile, NULL);
+    if (bRc != FALSE)
+    {
+        Fail("GetFileSizeEx: ERROR -> A file size was returned for "
+            "an invalid handle.\n");
+    }
+
+
+    /* create a test file */
+    hFile = CreateFile(szTextFile, 
+        GENERIC_READ | GENERIC_WRITE,
+        FILE_SHARE_READ | FILE_SHARE_WRITE,
+        NULL,
+        CREATE_ALWAYS,
+        FILE_ATTRIBUTE_NORMAL,
+        NULL);
+
+    if(hFile == INVALID_HANDLE_VALUE)
+    {
+        Fail("GetFileSizeEx: ERROR -> Unable to create file \"%s\".\n", 
+            szTextFile);
+    }
+
+    /* give the file a size */
```

Comment:
I haven't noticed that before. Seems like it would be a nice cleanup, so if you want to do that, then go ahead.